prompts/github.go

```diff
@@ -2,76 +2,171 @@ package prompts
 
 import (
 	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
 
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/huh"
+	"github.com/go-git/go-git/v5"
+	git_config "github.com/go-git/go-git/v5/config"
 	config "github.com/speakeasy-api/sdk-gen-config"
+	"github.com/speakeasy-api/sdk-gen-config/workflow"
+	"github.com/speakeasy-api/speakeasy/internal/charm"
 )
 
 const (
 	defaultGithubTokenSecretName     = "GITHUB_TOKEN"
 	defaultSpeakeasyAPIKeySecretName = "SPEAKEASY_API_KEY"
 )
 
-func githubWorkflowBaseForm(quickstart *Quickstart) (*QuickstartState, error) {
-	var targetType, specName string
-	for _, target := range quickstart.WorkflowFile.Targets {
-		targetType = target.Target
-		specName = target.Source
-		break
-	}
-
-	secrets := make(map[string]string)
-	secrets[config.GithubAccessToken] = formatSecretName(defaultGithubTokenSecretName)
-	secrets[config.SpeakeasyApiKey] = formatSecretName(defaultSpeakeasyAPIKeySecretName)
-
-	genWorkflow := &config.GenerateWorkflow{
-		Name: "Generate",
-		On: config.GenerateOn{
-			WorkflowDispatch: config.WorkflowDispatch{
-				Inputs: config.Inputs{
-					Force: config.Force{
-						Description: "Force generation of SDKs",
-						Type:        "boolean",
-						Default:     false,
+func ConfigureGithub(githubWorkflow *config.GenerateWorkflow, workflow *workflow.Workflow) (*config.GenerateWorkflow, error) {
+	if githubWorkflow == nil || githubWorkflow.Jobs.Generate.Uses == "" {
+		secrets := make(map[string]string)
+		secrets[config.GithubAccessToken] = formatSecretName(defaultGithubTokenSecretName)
+		secrets[config.SpeakeasyApiKey] = formatSecretName(defaultSpeakeasyAPIKeySecretName)
+		githubWorkflow = &config.GenerateWorkflow{
+			Name: "Generate",
+			On: config.GenerateOn{
+				WorkflowDispatch: config.WorkflowDispatch{
+					Inputs: config.Inputs{
+						Force: config.Force{
+							Description: "Force generation of SDKs",
+							Type:        "boolean",
+							Default:     false,
+						},
 					},
 				},
-			},
-			Schedule: []config.Schedule{
-				{
-					Cron: "0 0 * * *",
+				Schedule: []config.Schedule{
+					{
+						Cron: "0 0 * * *",
+					},
 				},
 			},
-		},
-		Jobs: config.Jobs{
-			Generate: config.Job{
-				Uses: "speakeasy-api/sdk-generation-action/.github/workflows/sdk-generation.yaml@v14",
-				With: map[string]any{
-					"speakeasy_version": "latest",
-					"force":             "${{ github.event.inputs.force }}",
-					config.OpenAPIDocs:  fmt.Sprintf("- %s\n", quickstart.WorkflowFile.Sources[specName].Inputs[0].Location),
-					config.Mode:         "pr",
-					config.Languages:    fmt.Sprintf("- %s\n", targetType),
+			Jobs: config.Jobs{
+				Generate: config.Job{
+					Uses: "speakeasy-api/sdk-generation-action/.github/workflows/sdk-generation.yaml@v15",
```

Comment:
Don't think we can ship this yet, right?

Reply:
Certainly not as is, because there is no action v15 in that repo

Reply:
probably not unless we point it to your temp repo